ote/src/clj/ote/transit_changes/detection.clj

```diff
@@ -838,6 +881,30 @@
   [db service-id route-list-with-changed-weeks]
   (mapv #(route-day-changes db service-id %) route-list-with-changed-weeks))
 
+(defn remove-outscoped-weeks
+  "Input: all-routes = sequence of vectors (template: `[ ['' {}] ['' {}] ]` ). Each vector describes a route.
+    routes-weeks = sequence of vectors. A vector contains maps of a route, a map describes a week of traffic.
+  Output: Removes from routes-weeks weeks which fall out from route's min and max date of traffic."
+  [all-routes routes-weeks]
+  (mapv
+    (fn [route-wks]
+      ;; Take route-info from first object with matching route key. Same in all because a vector contains one route.
+      (let [route-info (some #(when (= (first (keys (:routes (first route-wks))))
+                                       (first %))
+                                (second %))
+                             all-routes)
+            route-min-date-local (when route-info
+                                   (.toLocalDate (:min-date route-info)))
+            route-max-date-local (when route-info
+                                   (.toLocalDate (:max-date route-info)))]
+
+        (if (and route-min-date-local route-max-date-local)
+          (filterv #(and (.isAfter (:end-of-week %) route-min-date-local)
+                         (.isBefore (:beginning-of-week %) route-max-date-local))
```

Comment:
I'm assuming that localdate don't have time (hours) so when getting isBefore :beginning-of-week might be same day, but with time (hours) and the comparing works even if they are pointing to the same day?

Reply:
Good catch, will change.